src/RosettaPy/rosetta.py

```
@@ -92,33 +92,45 @@
-            if self.bin.mode != "mpi":
+            self.use_mpi = self.run_node.mpi_available
+
+        else:
+            # for the rest of node types
+            # repect to the final choice of user.
+            self.run_node.mpi_available = self.use_mpi
+
+        # warnings about the bin.mode not 'mpi' (None, for example) yet self.use_mpi == True
+        if self.bin.mode != "mpi":
+            if self.use_mpi:
                 warnings.warn(
```

Ruff

No explicit `stacklevel` keyword argument found

(B028)